src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -3465,17 +3466,25 @@ def set_lb_frontend_ip_configuration(
     return parent
 
 
+def _process_vnet_name_and_id(vnet, cmd, resource_group_name):
+    if vnet and not is_valid_resource_id(vnet):
+        vnet = resource_id(
+            subscription=get_subscription_id(cmd.cli_ctx),
+            resource_group=resource_group_name,
+            namespace='Microsoft.Network',
+            type='virtualNetworks',
+            name=vnet)
+    return vnet
+
+
+def _process_subnet_name_and_id(subnet, vnet, cmd, resource_group_name):
+    if subnet and not is_valid_resource_id(subnet):
+        subnet = _process_vnet_name_and_id(vnet, cmd, resource_group_name) + f'/subnets/{subnet}'
```

Comment:
what if _process_vnet_name_and_id return None?

Reply:
There is check before here.